Controller/SectionController.php

```diff
@@ -176,63 +101,40 @@ public function viewAction($sectionId)
     /**
      * Deletes a section.
      *
-     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param mixed $sectionId
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function deleteAction(Request $request)
+    public function deleteAction($sectionId)
     {
         try {
-            $sectionUpdateStruct = $this->sectionService->newSectionUpdateStruct();
-            $form = $this->generateDeleteForm($sectionUpdateStruct);
-            $form->handleRequest($request);
-
-            if ($form->isValid()) {
-                $section = $this->sectionService->loadSectionByIdentifier($sectionUpdateStruct->identifier);
-                $this->sectionService->deleteSection($section);
-            }
+            $section = $this->sectionService->loadSection($sectionId);
+            $this->sectionService->deleteSection($section);
         } catch (UnauthorizedException $e) {
             return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
         }
 
+        $this->notificationPool->addNotification(new TranslatableNotificationMessage([
+            'message' => 'section.deleted',
+            'translationParams' => ['%sectionName%' => $section->name],
+            'domain' => 'section',
+        ]));
+
         return $this->redirect($this->generateUrl('admin_sectionlist'));
     }
 
-    /**
-     * Displays the create form and processes it once submitted.
-     *
-     * @param \Symfony\Component\HttpFoundation\Request $request
-     *
-     * @return \Symfony\Component\HttpFoundation\Response
-     */
-    public function createAction(Request $request)
+    public function createAction()
     {
-        $section = new Section();
-
-        $form = $this->createForm($this->sectionType, $section, [
-            'action' => $this->router->generate('admin_sectioncreate'),
-        ]);
-
-        $form->handleRequest($request);
-
-        if ($form->isValid()) {
-            try {
-                $newSection = $this->sectionHelper->createSection($section);
-
-                return $this->redirect(
-                    $this->generateUrl(
-                        'admin_sectionview',
-                        ['sectionId' => $newSection->id]
-                    )
-                );
-            } catch (UnauthorizedException $e) {
-                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
-            } catch (InvalidArgumentException $e) {
-                $this->addAlreadyExistErrorMessage();
-            }
+        try {
+            $section = $this->sectionService->createSection(new SectionCreateStruct([
```

Comment:
by doing that, as soon as the user clicks on "Create a new section", the section is created before he validates the form. Creating something on the server with a GET request is bad (like deleting) and worse, if the user navigates elsewhere or closes his browser, the new section has been created but the uses has absolutely no clue about that.